netbox_onboarding/models.py

```diff
@@ -78,40 +80,67 @@ class OnboardingDevice(models.Model):
     @property
     def last_check_attempt_date(self):
         """Date of last onboarding attempt for a device."""
-        try:
-            return OnboardingTask.objects.filter(created_device=self.device).latest("created_on").created_on
-        except ValueError:
+        if self.device.primary_ip4:
+            try:
+                return (
+                    OnboardingTask.objects.filter(
+                        ip_address=self.device.primary_ip4.address.ip.format()  # pylint: disable=no-member
```

Comment:
Per Slack discussion, this is a ~bug fix~ *improvement*, as an OnboardingTask that failed due to an exception may have an `ip_address` but not a `created_device`, and we want to match such tasks in the filter.

Reply:
I updated MR message